hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -126,14 +127,24 @@ private AuthenticationStatus authenticate() {
 
     private AuthenticationStatus authenticate(SecurityContext securityContext) {
         Connection connection = endpoint.getConnection();
+        Boolean passed = Boolean.FALSE;
         try {
             LoginContext lc = securityContext.createClientLoginContext(clusterName, credentials, connection);
             lc.login();
             endpoint.setLoginContext(lc);
+            passed = Boolean.TRUE;
             return AUTHENTICATED;
         } catch (LoginException e) {
             logger.warning(e);
             return CREDENTIALS_FAILED;
+        } finally {
+            nodeEngine.getNode().getNodeExtension().getAuditlogService()
+                .eventBuilder(AuditlogTypeIds.AUTHENTICATION_CLIENT)
+                .message("Client connection authentication.")
+                .addParameter("connection", connection)
```

Comment:
Unfortunately seems params are an easy way how to leak private API. User implementations can now get references to classes such as `com.hazelcast.internal.nio.Connection`. I don't see an easy way out of it but I'd prefer that we don't do it.

Reply:
I wouldn't call it leaking private API. We don't do any promise about specific types. Params are just objects. If the consuming class doesn't recognize the type, it should just use `Object.toString()`.

Reply:
Just wondering, if a user implements some service and uses the methods on the `Connection` but we move the `Connection` class or remove it's methods in a patch release - is that ok? Is it something that we can warn users about? Like - you can downcast to specific parameter classes but it's still private API?

Reply:
The warning will be added to `EventBuilder` javadoc.